File: app/src/main/java/com/example/monappliimmo/MainActivity.java
package com.example.monappliimmo;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // On appelle la configuration de la navigation
        setupNavigation();
    }

    private void setupNavigation() {
        BottomNavigationView bottomNav = findViewById(R.id.bottom_navigation);

        // Écran par défaut (Home)
        chargerFragment(new HomeFragment());

        // Gestion des clics
        bottomNav.setOnItemSelectedListener(item -> {
            int id = item.getItemId();
            if (id == R.id.nav_appartements) {
                return chargerFragment(new HomeFragment());
            } else if (id == R.id.nav_batiments) {
                // return chargerFragment(new BatimentFragment());
            }
            return false;
        });
    }

    private boolean chargerFragment(Fragment fragment) {
        if (fragment != null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, fragment)
                    .commit();
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/example/monappliimmo/MainActivity.kt
package com.example.monappliimmo

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupNavigation()
    }

    private fun setupNavigation() {
        val bottomNav = findViewById<BottomNavigationView>(R.id.bottom_navigation)

        // Écran par défaut (Liste des appartements)
        chargerFragment(HomeFragment())

        bottomNav.setOnItemSelectedListener { item: MenuItem ->
            when (item.itemId) {

                R.id.nav_appartements -> {
                    chargerFragment(HomeFragment())
                    true
                }

                R.id.nav_batiments -> {
                    chargerFragment(BatimentFragment())
                    true
                }

                R.id.nav_interventions -> {
                    chargerFragment(InterventionFragment())
                    true
                }

                // Clic sur l'onglet "Compte" pour ouvrir ton superbe écran de login
                R.id.nav_compte -> {
                    val intent = Intent(this, LoginActivity::class.java)
                    startActivity(intent)
                    // On retourne false pour ne pas changer d'onglet visuellement
                    // car on change carrément de page (Activity)
                    false
                }

                else -> false
            }
        }
    }

    private fun chargerFragment(fragment: Fragment): Boolean {
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .commit()
        return true
    }
}